### PR TITLE
This makes the float errors explicit.

### PR DESCRIPTION
**Pull request by lemire**
If x and y differ as numbers, it is not true that their naive string representation will differ. We need to be careful.

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser This is meant to be merged into "jkeiser/ondemand-sajson"https://github.com/simdjson/simdjson/pull/1372  not into master. Please review and merge if you like it.

File: .github/workflows/vs16-noexcept-ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: VS16-NoExcept-CI
```
Comment (jkeiser) [inline .github/workflows/vs16-noexcept-ci.yml:1]: Was this intended?

Comment (lemire) [reply .github/workflows/vs16-noexcept-ci.yml:1]: Yes. It is like what we have in the main branch. The convention is to use at least one distinct name per file (i.e., if you have k iml files, you should have at least k distinct names).

Comment (jkeiser) [reply .github/workflows/vs16-noexcept-ci.yml:1]: Oh sorry, I was checking if you intended to add the file :)

Comment (lemire) [reply .github/workflows/vs16-noexcept-ci.yml:1]: Yes. This file is not really part of the PR. I only really changed the error reporting, and only for the double type.

**Pull request merged**